Create layout page for markdown pages

maribelferreira opened the pull request:
- Created new Markdown.astro layout with comprehensive styling
- Updated architecture, prior art, spec, and UX pages to use new layout
- Added descriptions to frontmatter
- Removed nav_order from pages
- Updated image path for architecture diagram
- Added Tailwind Typography plugin to global styles

# What kind of change does this PR introduce?
 
<!-- (Bug fix, feature, docs update, ...) -->

# What is the current behavior?
 
<!-- You can also link to an open issue here -->

# What is the new behavior (if this is a feature change)?

# Other information

maribelferreira committed:
Add Markdown layout for documentation pages

- Created new Markdown.astro layout with comprehensive styling
- Updated architecture, prior art, spec, and UX pages to use new layout
- Added descriptions to frontmatter
- Removed nav_order from pages
- Updated image path for architecture diagram
- Added Tailwind Typography plugin to global styles

maribelferreira committed:
Refine Markdown layout typography and code styling

- Adjusted heading margins and line height for improved readability
- Simplified code block and inline code styling
- Updated text colors for better contrast
- Removed code block border and background
- Removed backtick pseudo-elements for inline code

maribelferreira committed:
Update documentation page metadata with more descriptive titles and detailed descriptions to be more SEO friendly

gnarea commented on src/pages/ux.md:
```diff
@@ -1,7 +1,8 @@
 ---
-title: "User Experience"
+title: "VeraId - User Experience"
+description: "Comprehensive guide to VeraId user roles and workflows, including organization administration, member management, and signature verification processes"
```
```suggestion
description: "Comprehensive guide to VeraId user roles and workflows, including organisation administration, member management, and signature verification processes"
```

gnarea reviewed (changes requested):
LGTM, except for a few minor things.

Instead of having to prefix `VeraId -` to each page title, I think we should just update the skeleton to add ` | VeraId` to the end.

maribelferreira committed:
Update src/pages/ux.md

Co-authored-by: Gus Narea <gnarea@users.noreply.github.com>

maribelferreira committed:
Update page titles in Skeleton layout and documentation pages

- Modified Skeleton.astro to append "| VeraId" to page titles
- Simplified page titles in markdown files by removing "VeraId -" prefix

maribelferreira committed:
- Added network-enabled dev script to package.json
- Reduced horizontal padding from px-8 to px-6
- Decreased top margin for h2 headings from mt-14 to mt-12

gnarea commented on src/layouts/Skeleton.astro:
```diff
@@ -9,6 +9,7 @@ interface Props {
 
 const { title, description } = Astro.props;
 const IS_PRODUCTION = import.meta.env.PROD;
+const pageTitle = `${title} | VeraId`;
```
I think it's a bit of an overkill to create a variable for this, when we won't be using it anywhere else.

gnarea reviewed (changes requested):
Sorry, one last thing I missed: I don't think it makes sense to call a layout `Makrdown`, just like we wouldn't call a layout `HTML`. I think `Page`, like we did in awala.app, is more appropriate.

maribelferreira committed:
Changed the name of the 'Markdown' page to 'Page'

maribelferreira committed:
Fix TopNav active link highlighting for trailing slash URLs

maribelferreira committed:
Simplify page title generation in Skeleton layout
